go.mod

```diff
@@ -6,237 +6,242 @@ require (
 	github.com/RHsyseng/operator-utils v0.0.0-20200213165520-1a022eb07a43
 	github.com/coreos/go-systemd v0.0.0-20191104093116-d3cd4ed1dbcf
 	github.com/coreos/ignition v0.35.0
-	github.com/coreos/ignition/v2 v2.18.0
+	github.com/coreos/ignition/v2 v2.17.0
 	github.com/go-logr/stdr v1.2.2
 	github.com/google/go-cmp v0.6.0
 	github.com/jaypipes/ghw v0.8.1-0.20210605191321-eb162add542b
 	github.com/kevinburke/go-bindata v3.16.0+incompatible
-	github.com/onsi/ginkgo/v2 v2.17.1
-	github.com/onsi/gomega v1.33.0
+	github.com/onsi/ginkgo/v2 v2.15.0
+	github.com/onsi/gomega v1.31.1
 	github.com/openshift-kni/debug-tools v0.1.12
-	github.com/openshift-kni/k8sreporter v1.0.5
-	github.com/openshift/api v0.0.0-20240422085825-2624175e9673
-	github.com/openshift/build-machinery-go v0.0.0-20231128094528-1e9b1b0595c8
-	github.com/openshift/client-go v0.0.0-20240415214935-be70f772f157
+	github.com/openshift-kni/k8sreporter v1.0.4
+	github.com/openshift/api v0.0.0-20240404200104-96ed2d49b255
+	github.com/openshift/build-machinery-go v0.0.0-20230816154005-5a38e1bfd880
+	github.com/openshift/client-go v0.0.0-20240325151210-66398bfc44e1
 	github.com/openshift/custom-resource-status v1.1.3-0.20220503160415-f2fdb4999d87
-	github.com/openshift/library-go v0.0.0-20240419113445-f1541d628746
+	github.com/openshift/hypershift/api v0.0.0-20240410070639-fabd790a6b09
+	github.com/openshift/library-go v0.0.0-20240404133914-6b235e9156fa
 	github.com/pkg/errors v0.9.1
-	github.com/prometheus/client_golang v1.19.0
+	github.com/prometheus/client_golang v1.18.0
 	github.com/sirupsen/logrus v1.9.3
 	github.com/spf13/cobra v1.8.0
 	github.com/spf13/pflag v1.0.6-0.20210604193023-d5e0c0615ace
 	gopkg.in/fsnotify.v1 v1.4.7
 	gopkg.in/ini.v1 v1.67.0
-	k8s.io/api v0.29.4
-	k8s.io/apiextensions-apiserver v0.29.4
-	k8s.io/apimachinery v0.29.4
-	k8s.io/client-go v0.29.4
-	k8s.io/code-generator v0.29.4
+	k8s.io/api v0.29.2
+	k8s.io/apiextensions-apiserver v0.29.2
+	k8s.io/apimachinery v0.29.2
+	k8s.io/client-go v12.0.0+incompatible
+	k8s.io/code-generator v0.29.2
 	k8s.io/klog v1.0.0
 	k8s.io/klog/v2 v2.120.1
-	k8s.io/kubelet v0.29.4
-	k8s.io/kubernetes v1.29.4
-	k8s.io/utils v0.0.0-20240310230437-4693a0247e57
+	k8s.io/kubelet v0.29.2
+	k8s.io/kubernetes v1.29.2
+	k8s.io/utils v0.0.0-20240102154912-e7106e64919e
 	kubevirt.io/qe-tools v0.1.8
-	sigs.k8s.io/controller-runtime v0.17.3
+	sigs.k8s.io/cluster-api v1.6.2
+	sigs.k8s.io/controller-runtime v0.17.2
```

Comment:
Looks like there is a downgrade of some of the components , is that intentional ?  If so , then what is the reason ?

Reply:
The is was the artifacts of `go mod tidy` + `go mod vendor` 
this is a minor version change so it doesn't contain breaking changes:
https://github.com/kubernetes-sigs/controller-runtime/compare/v0.17.2...v0.17.3

Reply:
what was the manual change ? (its obscured by go mod tidy and there are more "downgrades")

Reply:
If its due to https://github.com/kubernetes-sigs/controller-runtime/compare/v0.17.2...v0.17.3 diff,  will we be able to upgrade later once we can align to the changes ?